[GLIB] Disable hinting for testing

obyknovenius opened the pull request:
#### 34e13090c6050e91fd0b20bdab40f1dbcf404520
<pre>
[GLIB] Disable hinting for testing
<a href="https://bugs.webkit.org/show_bug.cgi?id=251912">https://bugs.webkit.org/show_bug.cgi?id=251912</a>

Reviewed by Carlos Garcia Campos.

There are some tests that differ between GTK and WPE because GTK
`WebKitTestRunner` disables font hinting. It does it by using
`GtkSettingsManager` mechanism which doesn&apos;t exist on WPE.

This patch introduces a new boolean WebProcess creation parameter
`disableFontHintingForTesting`. When set by `WebKitTestRunner`,
`CAIRO_HINT_METRICS_OFF` and `CAIRO_HINT_STYLE_NONE` will be applied
to `defaultCairoFontOptions`, and all subsequent calls of
`setDefaultCairoHintOptions()` will do nothing.

It let us reuse the same baselines for some tests.

* LayoutTests/platform/glib/accessibility/math-multiscript-attributes-expected.txt: Renamed from LayoutTests/platform/gtk/accessibility/math-multiscript-attributes-expected.txt.
* LayoutTests/platform/glib/css1/font_properties/font-expected.txt: Renamed from LayoutTests/platform/gtk/css1/font_properties/font-expected.txt.
* LayoutTests/platform/glib/css1/pseudo/multiple_pseudo_elements-expected.txt: Renamed from LayoutTests/platform/gtk/css1/pseudo/multiple_pseudo_elements-expected.txt.
* LayoutTests/platform/glib/css2.1/t051201-c23-first-line-00-b-expected.txt: Renamed from LayoutTests/platform/gtk/css2.1/t051201-c23-first-line-00-b-expected.txt.
* LayoutTests/platform/glib/css2.1/t1508-c527-font-04-b-expected.txt: Renamed from LayoutTests/platform/gtk/css2.1/t1508-c527-font-04-b-expected.txt.
* LayoutTests/platform/glib/css2.1/t1508-c527-font-05-b-expected.txt: Renamed from LayoutTests/platform/gtk/css2.1/t1508-c527-font-05-b-expected.txt.
* LayoutTests/platform/glib/css2.1/t1508-c527-font-07-b-expected.txt: Renamed from LayoutTests/platform/gtk/css2.1/t1508-c527-font-07-b-expected.txt.
* LayoutTests/platform/glib/css2.1/t1508-c527-font-10-c-expected.txt: Renamed from LayoutTests/platform/gtk/css2.1/t1508-c527-font-10-c-expected.txt.
* LayoutTests/platform/glib/fast/block/lineboxcontain/block-glyphs-replaced-expected.txt: Renamed from LayoutTests/platform/gtk/fast/block/lineboxcontain/block-glyphs-replaced-expected.txt.
* LayoutTests/platform/glib/fast/block/lineboxcontain/glyphs-expected.txt: Renamed from LayoutTests/platform/gtk/fast/block/lineboxcontain/glyphs-expected.txt.
* LayoutTests/platform/glib/fast/multicol/tall-image-behavior-expected.txt: Renamed from LayoutTests/platform/gtk/fast/multicol/tall-image-behavior-expected.txt.
* LayoutTests/platform/glib/fast/text/international/vertical-text-glyph-test-expected.txt: Renamed from LayoutTests/platform/gtk/fast/text/international/vertical-text-glyph-test-expected.txt.
* LayoutTests/platform/glib/fast/text/international/vertical-text-metrics-test-expected.txt: Renamed from LayoutTests/platform/gtk/fast/text/international/vertical-text-metrics-test-expected.txt.
* LayoutTests/platform/glib/fast/writing-mode/border-vertical-lr-expected.txt: Renamed from LayoutTests/platform/gtk/fast/writing-mode/border-vertical-lr-expected.txt.
* LayoutTests/platform/glib/fast/writing-mode/japanese-lr-text-expected.txt: Renamed from LayoutTests/platform/gtk/fast/writing-mode/japanese-lr-text-expected.txt.
* LayoutTests/platform/glib/fast/writing-mode/japanese-rl-text-expected.txt: Renamed from LayoutTests/platform/gtk/fast/writing-mode/japanese-rl-text-expected.txt.
* LayoutTests/platform/glib/mathml/opentype/horizontal-expected.txt: Renamed from LayoutTests/platform/gtk/mathml/opentype/horizontal-expected.txt.
* LayoutTests/platform/glib/mathml/opentype/horizontal-munderover-expected.txt: Renamed from LayoutTests/platform/gtk/mathml/opentype/horizontal-munderover-expected.txt.
* LayoutTests/platform/glib/mathml/opentype/large-operators-expected.txt: Renamed from LayoutTests/platform/gtk/mathml/opentype/large-operators-expected.txt.
* LayoutTests/platform/glib/mathml/opentype/vertical-expected.txt: Renamed from LayoutTests/platform/gtk/mathml/opentype/vertical-expected.txt.
* LayoutTests/platform/glib/mathml/presentation/bug159513-expected.txt: Renamed from LayoutTests/platform/gtk/mathml/presentation/bug159513-expected.txt.
* LayoutTests/platform/glib/mathml/presentation/mo-stretch-expected.txt: Renamed from LayoutTests/platform/gtk/mathml/presentation/mo-stretch-expected.txt.
* LayoutTests/platform/glib/mathml/presentation/roots-expected.txt: Renamed from LayoutTests/platform/gtk/mathml/presentation/roots-expected.txt.
* LayoutTests/platform/glib/svg/batik/text/verticalText-expected.txt: Renamed from LayoutTests/platform/gtk/svg/batik/text/verticalText-expected.txt.
* LayoutTests/platform/glib/tables/mozilla/bugs/bug2886-2-expected.txt: Renamed from LayoutTests/platform/gtk/tables/mozilla/bugs/bug2886-2-expected.txt.
* LayoutTests/platform/wpe/accessibility/math-multiscript-attributes-expected.txt: Removed.
* LayoutTests/platform/wpe/css1/font_properties/font-expected.txt: Removed.
* LayoutTests/platform/wpe/css1/pseudo/multiple_pseudo_elements-expected.txt: Removed.
* LayoutTests/platform/wpe/css2.1/t051201-c23-first-line-00-b-expected.txt: Removed.
* LayoutTests/platform/wpe/css2.1/t1508-c527-font-04-b-expected.txt: Removed.
* LayoutTests/platform/wpe/css2.1/t1508-c527-font-05-b-expected.txt: Removed.
* LayoutTests/platform/wpe/css2.1/t1508-c527-font-07-b-expected.txt: Removed.
* LayoutTests/platform/wpe/css2.1/t1508-c527-font-10-c-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/block/lineboxcontain/block-glyphs-replaced-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/block/lineboxcontain/glyphs-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/multicol/tall-image-behavior-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/text/international/vertical-text-glyph-test-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/text/international/vertical-text-metrics-test-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/writing-mode/border-vertical-lr-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/writing-mode/japanese-lr-text-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/writing-mode/japanese-rl-text-expected.txt: Removed.
* LayoutTests/platform/wpe/mathml/opentype/horizontal-expected.txt: Removed.
* LayoutTests/platform/wpe/mathml/opentype/horizontal-munderover-expected.txt: Removed.
* LayoutTests/platform/wpe/mathml/opentype/large-operators-expected.txt: Removed.
* LayoutTests/platform/wpe/mathml/opentype/vertical-expected.txt: Removed.
* LayoutTests/platform/wpe/mathml/presentation/bug159513-expected.txt: Removed.
* LayoutTests/platform/wpe/mathml/presentation/mo-stretch-expected.txt: Removed.
* LayoutTests/platform/wpe/mathml/presentation/roots-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/batik/text/verticalText-expected.txt: Removed.
* LayoutTests/platform/wpe/tables/mozilla/bugs/bug2886-2-expected.txt: Removed.
* Source/WebCore/platform/graphics/cairo/CairoUtilities.cpp:
(WebCore::getDefaultCairoFontOptions):
(WebCore::setDefaultCairoFontOptionsForTesting):
* Source/WebCore/platform/graphics/cairo/CairoUtilities.h:
* Source/WebKit/Shared/WebProcessCreationParameters.cpp:
(WebKit::WebProcessCreationParameters::encode const):
(WebKit::WebProcessCreationParameters::decode):
* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/APIProcessPoolConfiguration.cpp:
(API::ProcessPoolConfiguration::copy):
* Source/WebKit/UIProcess/API/APIProcessPoolConfiguration.h:
* Source/WebKit/UIProcess/API/C/glib/WKContextConfigurationGlib.cpp: Added.
(WKContextConfigurationOverrideDefaultCairoFontOptionsForTesting):
(WKContextConfigurationSetOverrideDefaultCairoFontOptionsForTesting):
* Source/WebKit/UIProcess/API/C/glib/WKContextConfigurationGlib.h: Added.
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::platformInitializeWebProcess):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::generateContextConfiguration const):
* Tools/WebKitTestRunner/TestController.h:
* Tools/WebKitTestRunner/gtk/main.cpp:
(initializeGtkSettings):

Canonical link: <a href="https://commits.webkit.org/260443@main">https://commits.webkit.org/260443@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9059b7f35fbac93f1592fc7b9b84a65f73357eab

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108106 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17176 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40974 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117231 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116553 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111995 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18696 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8472 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100310 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113874 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/14111 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97198 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41904 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/96018 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28836 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/83578 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10058 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30184 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10780 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7086 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16206 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49775 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12361 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3924 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebCore/platform/graphics/cairo/CairoUtilities.cpp:
```diff
@@ -49,9 +49,13 @@
 namespace WebCore {
 
 static CairoUniquePtr<cairo_font_options_t> s_defaultCairoFontOptions;
+static CairoUniquePtr<cairo_font_options_t> s_defaultCairoFontOptionsForTesting;
```
Why do we need this? Can't we just set s_defaultCairoFontOptions instead?

obyknovenius replied:
Another variable is required because on GTK `s_defaultCairoFontOptions` will be overwritten by `GtkSettingsManagerProxy`:
https://github.com/WebKit/WebKit/blob/a9febb26c871be962432254ecb980a9a36ac7f13/Source/WebKit/WebProcess/gtk/GtkSettingsManagerProxy.cpp#L170

carlosgcampos replied:
That's confusing, in the case of GTK since we set specific settings via GtkSettings, we could just disable hinting that way. no?

obyknovenius replied:
Yes, `GtkSettings` would work too. In fact, this is how it works now. I wanted to have one more explicit way to disable font hinting for testing which can be used on both platforms. I updated my PR. It doesn't require an additional variable only for testing. Moreover, `s_defaultCairoFontOptions` is always allocated only once. Font settings, such as hinting or antialiasing, are now applied separately from each other.

carlosgcampos commented on Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
```diff
@@ -167,6 +165,12 @@ void WebProcess::platformInitializeWebProcess(WebProcessCreationParameters& para
+    if (parameters.overrideDefaultCairoFontOptionsForTesting) {
+        CairoUniquePtr<cairo_font_options_t> fontOptions(cairo_font_options_create());
+        cairo_font_options_set_hint_style(fontOptions.get(), CAIRO_HINT_STYLE_NONE);
+        setDefaultCairoFontOptionsForTesting(WTFMove(fontOptions));
```
Why don't we call `setDefaultCairoFontOptions() ` this will always be called after the main where the previous call is done.

obyknovenius replied:
Same reason.

carlosgcampos commented on Tools/WebKitTestRunner/TestController.h:
```diff
@@ -45,6 +45,10 @@
+#if PLATFORM(GTK) || PLATFORM(WPE)
+#include "WKContextConfigurationGlib.h"
+#endif
+
```
Why do we need this in the header?

obyknovenius replied:
`TestController` calls `WKContextConfigurationSetOverrideDefaultCairoFontOptionsForTesting()` function which is defined in a separate GLIB-specific header `WKContextConfigurationGlib.h`. That's why this include is needed.

obyknovenius replied:
I could move it to `WKContextConfigurationRef.h` and surround it with `#if PLATFORM(GTK) || PLATFORM(WPE)`. I'm not sure which approach is better. I see that PlayStation port also adds some API using a separate header and did the same.

carlosgcampos replied:
Can't it be moved to TestController.cpp?

obyknovenius replied:
I've moved these GLIB-specific functions to `WKContextConfigurationRef.h` to the other `WKContextConfiguration*` functions. No additional headers is needed anymore.

obyknovenius replied:
That didn't work. Moving it back to a separate header. And, as you suggested, include it in `TestController.cpp`.

obyknovenius committed:
[GLIB] Disable hinting for testing
https://bugs.webkit.org/show_bug.cgi?id=251912

Reviewed by Carlos Garcia Campos.

There are some tests that differ between GTK and WPE because GTK
`WebKitTestRunner` disables font hinting. It does it by using
`GtkSettingsManager` mechanism which doesn't exist on WPE.

This patch introduces a new boolean WebProcess creation parameter
`disableFontHintingForTesting`. When set by `WebKitTestRunner`,
`CAIRO_HINT_METRICS_OFF` and `CAIRO_HINT_STYLE_NONE` will be applied
to `defaultCairoFontOptions`, and all subsequent calls of
`setDefaultCairoHintOptions()` will do nothing.

It let us reuse the same baselines for some tests.

* LayoutTests/platform/glib/accessibility/math-multiscript-attributes-expected.txt: Renamed from LayoutTests/platform/gtk/accessibility/math-multiscript-attributes-expected.txt.
* LayoutTests/platform/glib/css1/font_properties/font-expected.txt: Renamed from LayoutTests/platform/gtk/css1/font_properties/font-expected.txt.
* LayoutTests/platform/glib/css1/pseudo/multiple_pseudo_elements-expected.txt: Renamed from LayoutTests/platform/gtk/css1/pseudo/multiple_pseudo_elements-expected.txt.
* LayoutTests/platform/glib/css2.1/t051201-c23-first-line-00-b-expected.txt: Renamed from LayoutTests/platform/gtk/css2.1/t051201-c23-first-line-00-b-expected.txt.
* LayoutTests/platform/glib/css2.1/t1508-c527-font-04-b-expected.txt: Renamed from LayoutTests/platform/gtk/css2.1/t1508-c527-font-04-b-expected.txt.
* LayoutTests/platform/glib/css2.1/t1508-c527-font-05-b-expected.txt: Renamed from LayoutTests/platform/gtk/css2.1/t1508-c527-font-05-b-expected.txt.
* LayoutTests/platform/glib/css2.1/t1508-c527-font-07-b-expected.txt: Renamed from LayoutTests/platform/gtk/css2.1/t1508-c527-font-07-b-expected.txt.
* LayoutTests/platform/glib/css2.1/t1508-c527-font-10-c-expected.txt: Renamed from LayoutTests/platform/gtk/css2.1/t1508-c527-font-10-c-expected.txt.
* LayoutTests/platform/glib/fast/block/lineboxcontain/block-glyphs-replaced-expected.txt: Renamed from LayoutTests/platform/gtk/fast/block/lineboxcontain/block-glyphs-replaced-expected.txt.
* LayoutTests/platform/glib/fast/block/lineboxcontain/glyphs-expected.txt: Renamed from LayoutTests/platform/gtk/fast/block/lineboxcontain/glyphs-expected.txt.
* LayoutTests/platform/glib/fast/multicol/tall-image-behavior-expected.txt: Renamed from LayoutTests/platform/gtk/fast/multicol/tall-image-behavior-expected.txt.
* LayoutTests/platform/glib/fast/text/international/vertical-text-glyph-test-expected.txt: Renamed from LayoutTests/platform/gtk/fast/text/international/vertical-text-glyph-test-expected.txt.
* LayoutTests/platform/glib/fast/text/international/vertical-text-metrics-test-expected.txt: Renamed from LayoutTests/platform/gtk/fast/text/international/vertical-text-metrics-test-expected.txt.
* LayoutTests/platform/glib/fast/writing-mode/border-vertical-lr-expected.txt: Renamed from LayoutTests/platform/gtk/fast/writing-mode/border-vertical-lr-expected.txt.
* LayoutTests/platform/glib/fast/writing-mode/japanese-lr-text-expected.txt: Renamed from LayoutTests/platform/gtk/fast/writing-mode/japanese-lr-text-expected.txt.
* LayoutTests/platform/glib/fast/writing-mode/japanese-rl-text-expected.txt: Renamed from LayoutTests/platform/gtk/fast/writing-mode/japanese-rl-text-expected.txt.
* LayoutTests/platform/glib/mathml/opentype/horizontal-expected.txt: Renamed from LayoutTests/platform/gtk/mathml/opentype/horizontal-expected.txt.
* LayoutTests/platform/glib/mathml/opentype/horizontal-munderover-expected.txt: Renamed from LayoutTests/platform/gtk/mathml/opentype/horizontal-munderover-expected.txt.
* LayoutTests/platform/glib/mathml/opentype/large-operators-expected.txt: Renamed from LayoutTests/platform/gtk/mathml/opentype/large-operators-expected.txt.
* LayoutTests/platform/glib/mathml/opentype/vertical-expected.txt: Renamed from LayoutTests/platform/gtk/mathml/opentype/vertical-expected.txt.
* LayoutTests/platform/glib/mathml/presentation/bug159513-expected.txt: Renamed from LayoutTests/platform/gtk/mathml/presentation/bug159513-expected.txt.
* LayoutTests/platform/glib/mathml/presentation/mo-stretch-expected.txt: Renamed from LayoutTests/platform/gtk/mathml/presentation/mo-stretch-expected.txt.
* LayoutTests/platform/glib/mathml/presentation/roots-expected.txt: Renamed from LayoutTests/platform/gtk/mathml/presentation/roots-expected.txt.
* LayoutTests/platform/glib/svg/batik/text/verticalText-expected.txt: Renamed from LayoutTests/platform/gtk/svg/batik/text/verticalText-expected.txt.
* LayoutTests/platform/glib/tables/mozilla/bugs/bug2886-2-expected.txt: Renamed from LayoutTests/platform/gtk/tables/mozilla/bugs/bug2886-2-expected.txt.
* LayoutTests/platform/wpe/accessibility/math-multiscript-attributes-expected.txt: Removed.
* LayoutTests/platform/wpe/css1/font_properties/font-expected.txt: Removed.
* LayoutTests/platform/wpe/css1/pseudo/multiple_pseudo_elements-expected.txt: Removed.
* LayoutTests/platform/wpe/css2.1/t051201-c23-first-line-00-b-expected.txt: Removed.
* LayoutTests/platform/wpe/css2.1/t1508-c527-font-04-b-expected.txt: Removed.
* LayoutTests/platform/wpe/css2.1/t1508-c527-font-05-b-expected.txt: Removed.
* LayoutTests/platform/wpe/css2.1/t1508-c527-font-07-b-expected.txt: Removed.
* LayoutTests/platform/wpe/css2.1/t1508-c527-font-10-c-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/block/lineboxcontain/block-glyphs-replaced-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/block/lineboxcontain/glyphs-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/multicol/tall-image-behavior-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/text/international/vertical-text-glyph-test-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/text/international/vertical-text-metrics-test-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/writing-mode/border-vertical-lr-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/writing-mode/japanese-lr-text-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/writing-mode/japanese-rl-text-expected.txt: Removed.
* LayoutTests/platform/wpe/mathml/opentype/horizontal-expected.txt: Removed.
* LayoutTests/platform/wpe/mathml/opentype/horizontal-munderover-expected.txt: Removed.
* LayoutTests/platform/wpe/mathml/opentype/large-operators-expected.txt: Removed.
* LayoutTests/platform/wpe/mathml/opentype/vertical-expected.txt: Removed.
* LayoutTests/platform/wpe/mathml/presentation/bug159513-expected.txt: Removed.
* LayoutTests/platform/wpe/mathml/presentation/mo-stretch-expected.txt: Removed.
* LayoutTests/platform/wpe/mathml/presentation/roots-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/batik/text/verticalText-expected.txt: Removed.
* LayoutTests/platform/wpe/tables/mozilla/bugs/bug2886-2-expected.txt: Removed.
* Source/WebCore/platform/graphics/cairo/CairoUtilities.cpp:
(WebCore::getDefaultCairoFontOptions):
(WebCore::setDefaultCairoFontOptionsForTesting):
* Source/WebCore/platform/graphics/cairo/CairoUtilities.h:
* Source/WebKit/Shared/WebProcessCreationParameters.cpp:
(WebKit::WebProcessCreationParameters::encode const):
(WebKit::WebProcessCreationParameters::decode):
* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/APIProcessPoolConfiguration.cpp:
(API::ProcessPoolConfiguration::copy):
* Source/WebKit/UIProcess/API/APIProcessPoolConfiguration.h:
* Source/WebKit/UIProcess/API/C/glib/WKContextConfigurationGlib.cpp: Added.
(WKContextConfigurationOverrideDefaultCairoFontOptionsForTesting):
(WKContextConfigurationSetOverrideDefaultCairoFontOptionsForTesting):
* Source/WebKit/UIProcess/API/C/glib/WKContextConfigurationGlib.h: Added.
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::platformInitializeWebProcess):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::generateContextConfiguration const):
* Tools/WebKitTestRunner/TestController.h:
* Tools/WebKitTestRunner/gtk/main.cpp:
(initializeGtkSettings):

Canonical link: https://commits.webkit.org/260443@main
